File: php/listener.php
<?php
$token_correto = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123"; // O token individual desse cliente
$headers = getallheaders();
if (!isset($headers['X-Webhook-Token']) || $headers['X-Webhook-Token'] !== $token_correto) {
    http_response_code(403);
    exit("Acesso negado!");
}

$input = file_get_contents('php://input');
$data = json_decode($input, true);

$log = "---- ".date("Y-m-d H:i:s")." ----\n";
$log .= "INPUT RAW: $input\n";
$log .= "DECODED:\n".print_r($data, true)."\n";
$log .= "------------------------\n";
file_put_contents(__DIR__ . '/webhook_log.txt', $log, FILE_APPEND);

$assunto = "Webhook recebido!";
$mensagem = "Recebido:\n\n" . print_r($data, true);
$headers_mail  = "From: webhook@seudominio.com\r\n";
mail("alvi@alaj.com.br", $assunto, $mensagem, $headers_mail);

http_response_code(200);
echo "OK";
?>

File: python/listener.py
from flask import Flask, request, abort
import smtplib
from email.message import EmailMessage
import datetime

app = Flask(__name__)
TOKEN_CORRETO = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123"  # O token individual desse cliente

@app.route("/", methods=["POST"])
def receber_webhook():
    token = request.headers.get('X-Webhook-Token')
    if token != TOKEN_CORRETO:
        abort(403)
    data = request.get_json()
    # Grava log
    with open("webhook_log.txt", "a") as f:
        f.write(f"---- {datetime.datetime.now()} ----\nRAW: {request.data.decode()}\nDECODED: {data}\n------------------------\n")
    # Envia email
    msg = EmailMessage()
    msg.set_content("Recebido:\n\n" + str(data))
    msg['Subject'] = "Webhook recebido!"
    msg['From'] = "webhook@seudominio.com"
    msg['To'] = "alvi@alaj.com.br"
    with smtplib.SMTP("localhost") as s:
        s.send_message(msg)
    return "OK"

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)
